### ISSUE TITLE
per-user persistent meta-data

### ISSUE THREAD

**Issue by tacaswell** [bluesky/bluesky-queueserver#25]
We should be able to capture metadata from the user (not already in the sample or admin databases) and persist that meta-data across sessions.

**Comment by tacaswell** [comment]
The depends on #24 so we know who the user is.

**Comment by developingAlex** [comment]
Does this have anything to do with adding in custom metadata to the run engine when it runs a plan?
I was wondering if it would make sense to be able to take in an additional "meta" type key from the plan_parsed at the top of [this block](https://github.com/bluesky/bluesky-queueserver/blob/20e22731854935d351ca5c0a3ca49ef08691fb07/bluesky_queueserver/manager/worker.py#L168-L189) in the worker and pass in its keys into the call to the self._RE(...) so as to get that metadata back in the bluesky "start" document ?

**Comment by dmgav** [comment]
@developingAlex Yes. We are getting to the point when we will have to implement this. There are multiple ways to implement this. One way is to pass metadata as one of the plan parameters (e.g. all predefined Bluesky plans accept parameter `md`, which is a dictionary that contains arbitrary metadata). We can definitely implement an additional `meta` key. This may be preferred approach. Persistent metadata (user/proposal data from the database) may be automatically appended to the contents of the dictionary before the plan is sent to RE Worker. I think the most important thing is to make a decision on exact metadata flow in the system. 

I think this issue is mostly related to web-client, since most of the work will be to develop GUI components for entering and editing metadata.
